apps/yggdrasil/lib/yggdrasil/user.ex

```diff
@@ -9,6 +10,7 @@ defmodule Yggdrasil.User do
     field :hash, :string
     field :password, :string, virtual: true # not part of table
     field :password_confirmation, :string, virtual: true # not part of table
+    belongs_to :role, Yggdrasil.Role
```

Comment:
I need to look up has one again may have confused them. I was thinking has one and has many is from the other side of the relationship. Since the user has a foreign key in it. 

Thought about multiple roles but couldn't think of a good use for it.

Off hand maybe player and admin together where admin just layers on admin permission?

I could see that possibly where you start with a base role and layer on permissions by adding roles? Then checking them would be flattening them out when checking resources and if any of them have the permission it passes.